### PR TITLE
Add  informergen

### PR DESCRIPTION
**Pull request by njhale**
Add the `informer` generator to generate kcp-aware wrappers for existing informers.

### CODE REVIEW

File: pkg/generators/informergen/gen.go [hunk 1]
```diff
@@ -0,0 +1,484 @@
...
+const (
+	// GeneratorName is the name of the generator.
+	GeneratorName = "informer"
+	// packageName for typed client wrappers.
+	typedPackageName = "externalversions"
+)
+
+type Generator struct {
+	// inputDir is the path where types are defined.
+	inputDir string
```
Comment (varshaprasad96) [inline pkg/generators/informergen/gen.go:49]: The generator would run a few additional fields to find the right path to scaffold out the informers. https://github.com/kcp-dev/code-generator/blob/3290e34c024601662f2000893aa03361a5a15121/pkg/generators/clientgen/gen.go#L73-L80, something like this to consider the cases of having apis in subdirectories. If needed, please feel free to move  helpers like [these](https://github.com/kcp-dev/code-generator/blob/3290e34c024601662f2000893aa03361a5a15121/pkg/generators/clientgen/gen.go#L155-L162) out of client-gen.

File: pkg/generators/informergen/gen.go [hunk 2]
```diff
@@ -0,0 +1,484 @@
...
+
+	if len(errs) > 0 {
+		return nil, loader.MaybeErrList(errs)
+	}
+
+	fmt.Printf("\tpackages: %v\n", pkgs)
+
+	// TODO: Find inputDir module separately.
+	var module string
+	for _, pkg := range pkgs {
```
Comment (varshaprasad96) [inline pkg/generators/informergen/gen.go:194]: `util.CurrentPacakge` could be helpful here.

**Comment by fabianvf** [comment]
in getting core types to work I seem to have broken the regular api imports, need to fix that

**Review by ncdc — CHANGES_REQUESTED** [review]
Can you add copyright headers to all new files that don't have them?

Are any of the new files copied from other repos, or are they all our brand new code?

File: examples/go.sum [hunk 3]
```diff
@@ -0,0 +1,658 @@
+cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
```
Comment (fabianvf) [inline examples/go.sum:1]: this was an accidental inclusion, will remove

**Comment by fabianvf** [comment]
> Can you add copyright headers to all new files that don't have them?

:facepalm: yep

> Are any of the new files copied from other repos, or are they all our brand new code?

The templates are _heavily inspired_ by existing informergen but not lifted directly

**Comment by ncdc** [comment]
(My force push was just rebasing on top of main and adjusting to namer being moved to third_party. No other changes.)

**Pull request merged**